src/dbnode/storage/flush.go

```diff
@@ -102,28 +105,73 @@ func (m *flushManager) Flush(
 
 	defer m.setState(flushManagerIdle)
 
-	// create flush-er
-	flushPersist, err := m.pm.StartFlushPersist()
+	namespaces, err := m.database.GetOwnedNamespaces()
 	if err != nil {
 		return err
 	}
 
-	namespaces, err := m.database.GetOwnedNamespaces()
+	// Perform three separate loops through all the namespaces so that we can
+	// emit better gauges, i.e. all the flushing for all the namespaces happens
+	// at once, then all the compaction, then all the snapshotting. This is also
+	// slightly better semantically because flushing should take priority over
+	// compaction and snapshotting.
+	//
+	// In addition, we need to make sure that for any given shard/blockStart
+	// combination, we attempt a flush before a snapshot as the snapshotting
+	// process will attempt to snapshot any unflushed blocks which would be
+	// wasteful if the block is already flushable.
+	multiErr := xerrors.NewMultiError()
+	if err = m.flush(namespaces, tickStart, dbBootstrapStateAtTickStart); err != nil {
```

Comment:
Would be good to have `data` in these function names so:

`dataFlush`
`dataCompact`
`dataSnapshot`

I also wonder if it might be even good to go so far as

`dataWarmFlush`
`dataColdCompactAndFlush`
`dataSnapshot`

Reply:
+1